### PR TITLE
EZP-26193: Invalid image upload

### PR DESCRIPTION
**Pull request by StephaneDiot**
jira : https://jira.ez.no/browse/EZP-26193

## Description

The patch correct a bug giving the possibility to upload 'false' images ( like any file renamed with a valid image extension).

## Tests 
Manual and unit tests

### CODE REVIEW

**Comment by StephaneDiot** [comment]
ready for review @yannickroger  @dpobel

File: Resources/public/js/views/fields/ez-image-editview.js [hunk 1]
```diff
@@ -51,6 +51,55 @@ YUI.add('ez-image-editview', function (Y) {
         },
 
         /**
+         * Set the file attribute with a struct based on the file object from the
+         * input file element. The struct uri is also tested by a test image to see
```
Comment (dpobel) [inline Resources/public/js/views/fields/ez-image-editview.js]: The part about the test is not super clear. You could write something like:
```
This implementation adds a validation to check that the chosen file is really an image the browser can load.
```

File: Resources/public/js/views/fields/ez-image-editview.js [hunk 2]
```diff
@@ -51,6 +51,55 @@ YUI.add('ez-image-editview', function (Y) {
...
+         * if it is valid
+         *
+         * @method _base64ToFileStruct
+         * @protected
+         * @param {EventFacade} e event facade
+         * @param {File} file the File object from the input file element
+         */
+        _base64ToFileStruct: function (file, e) {
+            var base64 = e.target.result.replace(/^.*;base64,/, ''),
+                myImage = new Image(100, 200),
```
Comment (dpobel) [inline Resources/public/js/views/fields/ez-image-editview.js]: bad variable name, could be `testImageObject` or something like that.

File: Resources/public/js/views/fields/ez-image-editview.js [hunk 3]
```diff
@@ -51,6 +51,55 @@ YUI.add('ez-image-editview', function (Y) {
...
+                fileStruct = this._createFileStruct(file, base64);
+
+            myImage.src = fileStruct.displayUri;
+            myImage.addEventListener ("error", Y.bind(function() {
```
Comment (dpobel) [inline Resources/public/js/views/fields/ez-image-editview.js]: no space after `addEventListener` but one after `function` (same below)

File: Resources/public/js/views/fields/ez-image-editview.js [hunk 4]
```diff
@@ -51,6 +51,55 @@ YUI.add('ez-image-editview', function (Y) {
...
+            e.target.onload = undefined;
+        },
+
+        /**
+         * Removes the image update class
+         *
+         * @method _removeImageUpdate
+         * @protected
+         */
+        _removeImageUpdate: function () {
```
Comment (dpobel) [inline Resources/public/js/views/fields/ez-image-editview.js]: bad name, what is "Image Update" ? should be called `_removeImageBeingUpdatedClass` (and the doc comment should be fixed accordingly)

File: Tests/js/views/fields/assets/ez-image-editview-tests.js [hunk 5]
```diff
@@ -464,6 +464,115 @@ YUI.add('ez-image-editview-tests', function (Y) {
...
+                    eventFacade = new Y.DOMEventFacade({
+                        type: 'change'
+                    });
+
+                this.originalURL = win.URL;
+                win.URL = {};
+                win.URL.createObjectURL = function (uri) {};
+
+                eventFacade.target = new Mock();
+                Mock.expect(eventFacade.target, {
```
Comment (dpobel) [inline Tests/js/views/fields/assets/ez-image-editview-tests.js]: this is exactly the same as in the previous test. Same for the next Mock.expect call. And the 3rd one can easily be put in a method with a parameter.

File: Tests/js/views/fields/assets/ez-image-editview-tests.js [hunk 6]
```diff
@@ -464,6 +464,115 @@ YUI.add('ez-image-editview-tests', function (Y) {
...
+                });
+
+                this.view.render();
+                this.view._updateFile(eventFacade);
+                this.view.after('warningChange', this.next(Y.bind(function (e) {
+                    Assert.isString(
+                        this.view.get('warning'),
+                        "A warning should have been generated"
+                    );
+                    Assert.areNotSame(
```
Comment (dpobel) [inline Tests/js/views/fields/assets/ez-image-editview-tests.js]: are(Not)Same to compare 2 numbers is a bit weird. areNotEqual is more suited.

File: Tests/js/views/fields/assets/ez-image-editview-tests.js [hunk 7]
```diff
@@ -789,6 +898,71 @@ YUI.add('ez-image-editview-tests', function (Y) {
                     [{size: (this.maxSize - 1) * this.multiplicator, name: "wrong.ogg", type: "audio/ogg"}]
                 );
             },
+
+            "Should accept the dropped file": function () {
+                var dropArea,
```
Comment (dpobel) [inline Tests/js/views/fields/assets/ez-image-editview-tests.js]: it's almost a complete copy paste of the method in the "parent" object...

File: Tests/js/views/fields/assets/ez-image-editview-tests.js (hunk 5 above)
Comment (StephaneDiot) [inline Tests/js/views/fields/assets/ez-image-editview-tests.js]: I'm not overriding binaryfile test cases anymore so now it's the only test using this mocks. That's why I believe I can now keep them where they are.

File: Resources/public/js/views/fields/ez-image-editview.js [hunk 8]
```diff
@@ -51,6 +51,55 @@ YUI.add('ez-image-editview', function (Y) {
...
+         * @protected
+         * @param {EventFacade} e event facade
+         * @param {File} file the File object from the input file element
+         */
+        _base64ToFileStruct: function (file, e) {
+            var base64 = e.target.result.replace(/^.*;base64,/, ''),
+                testImageObject = new Image(100, 200),
+                fileStruct = this._createFileStruct(file, base64);
+
+            testImageObject.src = fileStruct.displayUri;
```
Comment (dpobel) [inline Resources/public/js/views/fields/ez-image-editview.js]: you should probably move that line after setting the error and load event handlers.

File: Resources/public/js/views/fields/ez-image-editview.js [hunk 9]
```diff
@@ -51,6 +51,55 @@ YUI.add('ez-image-editview', function (Y) {
...
+            },this));
+            testImageObject.addEventListener("load", Y.bind(function () {
+                this._set('file', fileStruct);
+            }, this));
+
+            e.target.onload = undefined;
+        },
+
+        /**
+         * Removes the image update class
```
Comment (dpobel) [inline Resources/public/js/views/fields/ez-image-editview.js]: *the image being updated class*

**Comment by StephaneDiot** [comment]
ping @yannickroger

File: Tests/js/views/fields/assets/binarybase-tests.js [hunk 10]
```diff
@@ -563,6 +564,35 @@ YUI.add('binarybase-tests', function (Y) {
             };
         },
 
+        _readFileAssert: function (fileContent, file, binaryfile, fileReader) {
+            Assert.isFalse(
+                this.view.get('warning'),
+                "The warning attribute should stay false"
+            );
+
```
Comment (yannickroger) [inline Tests/js/views/fields/assets/binarybase-tests.js]: nitpick (I am almost feeling sorry), empty line

**Pull request merged**